core/unit_test/TestHalfConversion.hpp

```diff
@@ -35,16 +37,18 @@ void test_half_conversion_type() {
       });
 
   Kokkos::deep_copy(b, b_v);
-  ASSERT_LT((double(b - base) / double(base)), epsilon);
+  ASSERT_NEAR(b, base, epsilon);
 }
 
 template <class T>
 void test_bhalf_conversion_type() {
-  double epsilon = KOKKOS_BHALF_T_IS_FLOAT ? 0.0000003 : 0.0003;
-  T base         = static_cast<T>(3.3);
+  // When truncating mantissa to 7bits (like b16), 3.3 becomes 3.296875
+  // 3.3 - 3.296875 < 3.125e-3, so conversion error should be smaller
+  double epsilon                  = KOKKOS_BHALF_T_IS_FLOAT ? 3e-7 : 3.125e-3;
```

Comment:
Here, the same computation with 1ulp gives `0.015625`. The current bound is a lot tighter because we get luck and truncation begins with 00 `(3.3 -> 11.010011 | 00110011001100110011001100110011001100110011)`.